### PR TITLE
Fix for 245

### PR DESCRIPTION
**Pull request by Anurag870**
Making changes as per issue #245 which is follow up of #242

### CODE REVIEW

**Comment by Anurag870** [comment]
Signed CLA and linked email.

**Comment by kevinoliver** [comment]
Thanks for the patch! It seems like this causes the util-app unit test to fail which you can see in the travis-ci link (https://api.travis-ci.org/v3/job/564437988/log.txt).

```
[info] - App: pass in bad args and expect usage *** FAILED ***
[info]   "Exception thrown in main on startup" did not contain "Error parsing flag "environment"" (AppTest.scala:74)
```

Please take a look, thanks.

**Comment by Anurag870** [comment]
Thanks @kevinoliver, fixed it.

File: util-app/src/main/scala/com/twitter/app/GlobalFlag.scala [hunk 1]
```diff
@@ -117,9 +117,18 @@ abstract class GlobalFlag[T] private[app] (
 object GlobalFlag {
 
   private[app] def get(flagName: String): Option[Flag[_]] = {
-    def tryMethod(className: String, methodName: String): Option[Flag[_]] =
+    val className = if (!flagName.endsWith("$")) flagName + "$" else flagName
+    try {
+      get(Class.forName(className))
+    } catch {
+        case _: ClassNotFoundException | _: NoSuchFieldException | _: IllegalArgumentException =>
```
Comment (yufangong) [inline util-app/src/main/scala/com/twitter/app/GlobalFlag.scala]: Do we need the NoSuchFieldException & IllegalArgumentException here?

File: util-app/src/main/scala/com/twitter/app/GlobalFlag.scala [hunk 2]
```diff
@@ -130,9 +139,8 @@ object GlobalFlag {
           None
```
Comment (yufangong) [inline util-app/src/main/scala/com/twitter/app/GlobalFlag.scala]: Should we remove ClassNotFoundException here?

File: util-app/src/main/scala/com/twitter/app/GlobalFlag.scala (hunk 1 above)
Comment (Anurag870) [reply util-app/src/main/scala/com/twitter/app/GlobalFlag.scala]: @yufangong Thanks, fixed it

**Comment by yufangong** [comment]
@Anurag870 This is merged at https://github.com/twitter/util/commit/22b99087ade627f4e23404b12968e3609fe36f1f. Thanks for your contribution!

**Pull request closed without merging**